packages/core/src/modules/basic-messages/BasicMessagesApi.ts

```diff
@@ -1,35 +1,59 @@
+import type { BasicMessageProtocol, V2BasicMessage } from './protocols'
 import type { BasicMessageRecord } from './repository/BasicMessageRecord'
 import type { Query } from '../../storage/StorageService'
 
 import { AgentContext } from '../../agent'
-import { MessageHandlerRegistry } from '../../agent/MessageHandlerRegistry'
 import { MessageSender } from '../../agent/MessageSender'
 import { OutboundMessageContext } from '../../agent/models'
+import { AriesFrameworkError } from '../../error'
 import { injectable } from '../../plugins'
 import { ConnectionService } from '../connections'
 
-import { BasicMessageHandler } from './handlers'
-import { BasicMessageService } from './services'
+import { BasicMessagesModuleConfig } from './BasicMessagesModuleConfig'
+import { BasicMessageRepository } from './repository'
+
+export interface BasicMessagesApi<BMPs extends BasicMessageProtocol[]> {
+  sendMessage(connectionId: string, message: string, parentThreadId?: string): Promise<BasicMessageRecord>
+
+  findAllByQuery(query: Query<BasicMessageRecord>): Promise<BasicMessageRecord[]>
+  getById(basicMessageRecordId: string): Promise<BasicMessageRecord>
+  getByThreadId(threadId: string): Promise<BasicMessageRecord>
+  deleteById(basicMessageRecordId: string): Promise<void>
+}
 
 @injectable()
-export class BasicMessagesApi {
-  private basicMessageService: BasicMessageService
+export class BasicMessagesApi<BMPs extends BasicMessageProtocol[]> implements BasicMessagesApi<BMPs> {
```

Comment:
Do you think it's needed to register the protocols dynamically in this case? It's consistent with issue and prove, but those have reasons to not support e.g. v1 (anoncreds only). 

But if both v1 and v2 are registered by default, there's no reason to not make it configurable i guess